### PR TITLE
restart nginx when nitro-enclaves-acm is restarted

### PR DESCRIPTION
**Pull request by meerd**
*Description of changes:*
To ensure consistent state, **nginx** should be restarted instead of just reloading its configuration when the **nitro-enclaves-acm** service is restarted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/vtok_agent/src/agent/mod.rs [hunk 1]
```diff
@@ -164,11 +165,12 @@ impl PostSyncAction {
         );
         match self {
             Self::ReloadNginx => Self::reload_nginx(options.force_start, options.reload_wait_ms),
+            Self::RestartNginx => Self::restart_nginx(options.reload_wait_ms),
             Self::ReloadHttpd => Self::reload_httpd(options.force_start, options.reload_wait_ms),
         }
     }
 
-    fn reload_nginx(force_start: bool, wait_ms: u64) {
+    fn reload_nginx(force_start: bool, _: u64) {
```
Comment (mariusknaust) [inline src/vtok_agent/src/agent/mod.rs:173]: What is the reason we keep the second argument at all?

**Review by mariusknaust — DISMISSED** [review]

File: src/vtok_agent/src/agent/mod.rs [hunk 2]
```diff
@@ -194,6 +196,34 @@ impl PostSyncAction {
...
+
+            debug!("Sending SIGQUIT to PID={}", pid);
+            signal::kill(unistd::Pid::from_raw(pid), signal::Signal::SIGTERM).unwrap_or_else(
+                |err| {
+                    error!("Error sending SIGTERM: {:?}", err);
+                },
+            );
+        } else {
+            info!("NGINX service is not running");
+        }
```
Comment (agraf) [inline src/vtok_agent/src/agent/mod.rs:220]: Do we actually need to send those signals if we're issuing a service restart anyway?

File: src/vtok_agent/src/agent/mod.rs [hunk 3]
```diff
@@ -194,6 +196,34 @@ impl PostSyncAction {
             });
     }
 
+    fn restart_nginx(wait_ms: u64) {
+        info!("Restarting NGINX.");
+
+        if let Ok(pid) = is_service_running("nginx.service") {
+            debug!("Sending SIGWINCH to PID={}", pid);
```
Comment (clupuishere) [inline src/vtok_agent/src/agent/mod.rs:203]: Please add some info in the commit description about why `SIGWINCH` and `SIGQUIT` signals are used in these changes and what effect they have when sending to nginx.

File: src/vtok_agent/src/agent/mod.rs (hunk 1 above)
Comment (meerd) [reply src/vtok_agent/src/agent/mod.rs:173]: This value comes from the configuration. Following the previous change, it is currently unused. However, we may need to reuse this value in the near future. Therefore, I'm keeping it as is for now.

File: src/vtok_agent/src/agent/mod.rs (hunk 2 above)
Comment (meerd) [reply src/vtok_agent/src/agent/mod.rs:220]: The first signal is still essential for gracefully shutting down worker processes. This approach, in my opinion, offers a better user experience.

File: src/vtok_agent/src/agent/mod.rs (hunk 3 above)
Comment (meerd) [reply src/vtok_agent/src/agent/mod.rs:203]: SIGQUIT was removed. Updated the last commit's description.

File: src/vtok_agent/src/agent/mod.rs (hunk 2 above)
Comment (meerd) [reply src/vtok_agent/src/agent/mod.rs]: I also have doubts about the necessity of the second signal. I have removed it and pushed a new revision.

**Review by agraf — APPROVED** [review]

**Review by mariusknaust — APPROVED** [review]

**Review by clupuishere — APPROVED** [review]

File: src/vtok_agent/src/util.rs [hunk 4]
```diff
@@ -74,3 +75,58 @@ pub fn create_dirs_for_file<P: AsRef<Path>>(file_path: P) -> Result<(), std::io:
...
+pub fn service_start(service_name: &str) -> Result<(), SystemdError> {
+    service_exec("start", service_name)
+}
+
+pub fn service_restart(service_name: &str) -> Result<(), SystemdError> {
+    service_exec("restart", service_name)
+}
+
+pub fn systemd_reload() -> Result<(), SystemdError> {
+    service_exec("daemon-reload", "-f")
```
Comment (eugkoira) [inline src/vtok_agent/src/util.rs:131]: what's `-f`? cant' find it in the docs

**Review by eugkoira — APPROVED** [review]

File: src/vtok_agent/src/util.rs (hunk 4 above)
Comment (meerd) [reply src/vtok_agent/src/util.rs:131]:
```
  -f --force          When enabling unit files, override existing symlinks
```

**Pull request merged**